### PR TITLE
Adds support for browsing storybooks from the app

### PR DESCRIPTION
**Pull request by orta**
This means we can move away from these root view controllers in the app entirely, letting the JS control a lot more of the Emission app's flow.

* Re-jiggers the back button to look inside the top view controller for other navigations
* Updates styled-components to the official 2.0 release, now that it's out

Skip New Tests - not really relevant for the dev side of this.<hr data-danger="yep"/>

### Tested on Device?

- [ ] @orta
- [ ] @ashfurrow

<details>
  <summary>How to get set up with this PR?</summary>
  <p>&nbsp;</p>
   <p><b>To run on your computer:</b></p>
<pre><code>git fetch origin pull/569/head:orta-569-checkout
git checkout orta-569-checkout
yarn install
cd example; pod install; cd ..
open -a Simulator
yarn start</code></pre>
   </p>
   <p>Then run <code>xcrun simctl launch booted net.artsy.Emission</code> once a the simulator has finished booting</p>
   <p><b>To run inside Eigen (prod or beta) or Emission (beta):</b> Shake the phone to get the Admin menu.</p>
   <p>If you see <i>"Use Staging React Env" </i> - click that and restart, then follow the next step.</p>
   <p>Click on <i>"Choose an RN build" </i> - then pick the one that says: "X,Y,Z"</p>
   <p>Note: this is a TODO for PRs, currently  you can only do it on master commits.</p>
</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/components/storybooks/__tests__/section_browser-tests.tsx`
- `src/lib/components/storybooks/__tests__/story_browser-tests.tsx`
- `src/lib/components/storybooks/__tests__/styles-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 #### Consignments
 
+- Users now can access stories from inside the app - orta
```
Comment (ashfurrow) [inline CHANGELOG.md]: Should this be prefixed to match the format from the other entries?

File: Example/Emission/ARRootViewController.m [hunk 2]
```diff
@@ -77,12 +78,21 @@ - (ARSectionData *)developersSection
...
+  return [self tappableCellDataWithTitle:@"Open Storybook" selection: ^{
     id viewController = [ARStorybookComponentViewController new];
     [self.navigationController pushViewController:viewController animated:YES];
   }];
 }
 
+- (ARCellData *)jumpToEndUserStorybooks
+{
+  return [self tappableCellDataWithTitle:@"Open Storybook Browser" selection: ^{
+    id viewController = [[ARComponentViewController alloc] initWithEmission: nil moduleName:@"StorybookBrowser" initialProperties: @{}];
```
Comment (ashfurrow) [inline Example/Emission/ARRootViewController.m]: `initWithEmission: nil` extra space there, just a nitpick.

File: CHANGELOG.md (hunk 1 above)
Comment (orta) [inline CHANGELOG.md]: I've only been doing that for `[dev]` vs end-user focused things

File: Example/Emission/ARRootViewController.m (hunk 2 above)
Comment (orta) [inline Example/Emission/ARRootViewController.m]: I'll take it

**Pull request merged**